auditors/capabilities/capabilities.go

```diff
@@ -73,45 +65,74 @@ func getOverrideLabel(capability string) string {
 	return overrideLabelPrefix + strings.Replace(strings.ToLower(capability), "_", "-", -1)
 }
 
-func auditContainerForCapability(container *k8stypes.ContainerV1, capability string, dropList []string) (auditResults []*kubeaudit.AuditResult) {
-	if isCapabilityAdded(container, capability) {
-		auditResult := &kubeaudit.AuditResult{
-			Name:     CapabilityAdded,
+func auditContainer(container *k8stypes.ContainerV1, capability string, addList []string) []*kubeaudit.AuditResult {
+	var auditResults []*kubeaudit.AuditResult
+
+	if isCapabilityInArray(capability, addList) {
+		return auditResults
+	}
+
+	if SecurityContextOrCapabilities(container) {
+		var message string
+
+		if IsCapabilityInAddList(container, capability) {
+			message = fmt.Sprintf("Capability \"%s\" added. It should be removed from the capability add list. If you need this capability, add an override label such as '%s: SomeReason'.", capability, override.GetContainerOverrideLabel(container.Name, getOverrideLabel(capability)))
+			auditResult := &kubeaudit.AuditResult{
+				Name:     CapabilityAdded,
+				Severity: kubeaudit.Error,
+				Message:  message,
+				PendingFix: &fixCapabilityAdded{
+					container:  container,
+					capability: capability,
+				},
+				Metadata: kubeaudit.Metadata{
+					"Container": container.Name,
+					"Metadata":  capability,
+				},
+			}
+			auditResults = append(auditResults, auditResult)
+		}
+	}
+	// We need the audit result to be nil for ApplyOverride to check for RedundantAuditorOverride errors
+
+	if len(auditResults) == 0 {
+		return []*kubeaudit.AuditResult{nil}
+	}
+
+	return auditResults
+}
+
+func auditContainerForDropAll(container *k8stypes.ContainerV1) *kubeaudit.AuditResult {
+	var message string
+
```

Comment:
This is a bit weird as the message variable on line 109 is a different variable whereas the one on line 124 is using this one. I think we could make them both instantiate (using `:=`) and get rid of this function-scoped variable.

Reply:
Agreed! I probably had something else in mind when I did that, and ended up leaving that function-scoped variable there. I'll remove it!